test/src/main/java/org/apache/accumulo/test/fate/FateOpsCommandsIT.java

```diff
@@ -592,6 +610,124 @@ protected void testFateDeleteCommand(FateStore<TestEnv> store, ServerContext sct
+    executeTest(this::testFatePrintAndSummaryCommandsWithInProgressTxns);
+  }
+
+  protected void testFatePrintAndSummaryCommandsWithInProgressTxns(FateStore<TestEnv> store,
+      ServerContext sctx) throws Exception {
+    // This test was written for an issue with the 'admin fate --print' and 'admin fate --summary'
+    // commands where ZK NoNodeExceptions could occur. These commands first get a list of the
+    // transactions and then probe for info on these transactions. If a transaction completes
+    // between getting the list and probing for info on that transaction, a NoNodeException would
+    // occur causing the cmd to fail. This test ensures that this problem has been fixed (if the
+    // tx no longer exists, it should just be ignored so the print/summary can complete).
+    FateStore<TestEnv> mockedStore;
+    boolean isUserStore = store.type().equals(FateInstanceType.USER);
+
+    if (isUserStore) {
+      // The NNE can only occur for META transactions. For USER transactions, the transactions are
+      // stored in a table. Transactions can still complete mid-print, but an exception will
+      // not be thrown/need to be ignored. The equivalent for USER transactions would be
+      // the transaction returns an UNKNOWN status. So, we will ensure transactions with
+      // UNKNOWN status' are included in the output and don't cause any errors.
+      Method listMethod = UserFateStore.class.getMethod("list");
+      mockedStore =
+          EasyMock.createMockBuilder(UserFateStore.class).withConstructor(ClientContext.class)
+              .withArgs(sctx).addMockedMethod(listMethod).addMockedMethod("read").createMock();
+    } else {
+      // This error was occurring in AdminUtil.getTransactionStatus(). One of the methods that is
+      // called which may throw the NNE is top(), so we will mock this method to sometimes throw a
+      // NNE and ensure it is handled/ignored within getTransactionStatus() and that the rest
+      // of the transactions are returned.
+      Method listMethod = MetaFateStore.class.getMethod("list");
+      mockedStore = EasyMock.createMockBuilder(MetaFateStore.class)
+          .withConstructor(String.class, ZooReaderWriter.class)
+          .withArgs(sctx.getZooKeeperRoot() + Constants.ZFATE, sctx.getZooReaderWriter())
+          .addMockedMethod(listMethod).addMockedMethod("read").createMock();
+    }
+
+    FateId tx1 = mockedStore.create();
+    FateId tx2 = mockedStore.create();
+    FateId tx3 = mockedStore.create();
+    // Mock list() to ensure same order every run
+    List<ReadOnlyFateStore.FateIdStatus> fateIdStatusList =
+        List.of(createFateIdStatus(tx1), createFateIdStatus(tx2), createFateIdStatus(tx3));
+    expect(mockedStore.list()).andReturn(fateIdStatusList.stream()).once();
+
+    ReadOnlyFateStore.ReadOnlyFateTxStore<TestEnv> mockedFateTxStore1, mockedFateTxStore2,
+        mockedFateTxStore3;
+    if (isUserStore) {
```

Comment:
Opened #4688 after looking at this test because it highlighted that the behavior of the two fate stores is different.  If get #4688 is merged then this test could possibly be simplified alot because it can assume a single behavior instead of accommodating two.

Reply:
yes, that is a nice change